src/test/java/uk/gov/pay/api/utils/mocks/CreateChargeResponseFromConnector.java

```diff
@@ -0,0 +1,176 @@
+            this.reference = reference;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withEmail(String email) {
+            this.email = email;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withPaymentProvider(String paymentProvider) {
+            this.paymentProvider = paymentProvider;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withGatewayTransactionId(String gatewayTransactionId) {
+            this.gatewayTransactionId = gatewayTransactionId;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withCreatedDate(String createdDate) {
+            this.createdDate = createdDate;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withLanguage(SupportedLanguage language) {
+            this.language = language;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withDelayedCapture(boolean delayedCapture) {
+            this.delayedCapture = delayedCapture;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withCorporateCardSurcharge(Long corporateCardSurcharge) {
+            this.corporateCardSurcharge = corporateCardSurcharge;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withTotalAmount(Long totalAmount) {
+            this.totalAmount = totalAmount;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withRefundSummary(RefundSummary refundSummary) {
+            this.refundSummary = refundSummary;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withSettlementSummary(SettlementSummary settlementSummary) {
+            this.settlementSummary = settlementSummary;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withCardDetails(CardDetails cardDetails) {
+            this.cardDetails = cardDetails;
+            return this;
+        }
+
+        public CreateChargeResponseFromConnector build() {
+            List.of(amount, chargeId, language, cardDetails, links).forEach(Objects::requireNonNull);
+            
+            return new CreateChargeResponseFromConnector(amount, chargeId, state, returnUrl, description, reference, email, 
+                    paymentProvider, gatewayTransactionId, createdDate, language, delayedCapture, corporateCardSurcharge, 
+                    totalAmount, refundSummary, settlementSummary, cardDetails, links, metadata);
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withLink(Map<String, ?> link) {
+            links.add(link);
+            return this;
+        }
+
+        public CreateChargeResponseFromConnectorBuilder withMetadata(Map<String, String> metadata) {
```

Comment:
I think should allow a Map that has values spanning `String`, `Boolean` or `Number` (I've been using `Map<String, Object>`.